qp_shotgun/kneaddata/kneaddata.py

```diff
@@ -0,0 +1,272 @@
+    reverse_seqs : list of str
+        The list of reverse seqs filepaths
+    map_file : str
+        The path to the mapping file
+    out_dir : str
+        The job output directory
+    parameters : dict
+        The command's parameters, keyed by parameter name
+
+    Returns
+    -------
+    list of str
+        The KneadData commands
+    list of str
+        The run prefixes used
+
+
+    Raises
+    ------
+    ValueError
+        If the rev is not an empty list and the same length than fwd seqs
+        The prefixes of the run_prefix don't match the file names
+
+    Notes
+    -----
+    """
+    # making sure the forward and reverse reads are in the same order
+
+    # we match filenames, samples, and run prefixes
+    samples = make_read_pairs_per_sample(forward_seqs, reverse_seqs, map_file)
+
+    cmds = []
+
+    param_string = format_kneaddata_params(parameters)
+    prefixes = []
+    for run_prefix, sample, f_fp, r_fp in samples:
+        prefixes.append(run_prefix)
+        if r_fp is None:
+            cmds.append('kneaddata --input "%s" --output "%s" '
+                        '--output-prefix "%s" %s' %
+                        (f_fp, join(out_dir, run_prefix),
+                         run_prefix, param_string))
+        else:
+            cmds.append('kneaddata --input "%s" --input "%s" --output "%s" '
+                        '--output-prefix "%s" %s' %
+                        (f_fp, r_fp, join(out_dir, run_prefix),
+                         run_prefix, param_string))
+
+    return cmds, prefixes
+
+
+def _run_commands(qclient, job_id, commands, msg):
+    for i, cmd in enumerate(commands):
+        qclient.update_job_step(job_id, msg % i)
+        std_out, std_err, return_value = system_call(cmd)
+        if return_value != 0:
+            error_msg = ("Error running KneadData:\nStd out: %s\nStd err: %s"
+                         "\n\nCommand run was:\n%s"
+                         % (std_out, std_err, cmd))
+            return False, error_msg
+
+    return True, ""
+
+
+def _per_sample_ainfo(out_dir, prefixes):
+    ainfo = []
+    for prefix in prefixes:
+        sam_out_dir = join(out_dir, prefix)
+        sam_a = [
+            ArtifactInfo('clean paired R1', 'per_sample_FASTQ',
+                         [(join(sam_out_dir, '%s_paired_1.fastq' % prefix),
+                          'per_sample_FASTQ')]),
+            ArtifactInfo('clean paired R2', 'per_sample_FASTQ',
+                         [(join(sam_out_dir, '%s_paired_2.fastq' % prefix),
+                          'per_sample_FASTQ')]),
+            ArtifactInfo('clean unpaired R1', 'per_sample_FASTQ',
+                         [(join(sam_out_dir, '%s_unmatched_1.fastq' % prefix),
+                          'per_sample_FASTQ')]),
+            ArtifactInfo('clean unpaired R2', 'per_sample_FASTQ',
+                         [(join(sam_out_dir, '%s_unmatched_2.fastq' % prefix),
+                          'per_sample_FASTQ')])]
+        ainfo += sam_a
```

Comment:
is the intent for an `append` or an `extend`? would it be possible to use the desired method instead of an overloaded in place operator?